eng/release/Scripts/PublishToBlobAccounts.ps1

```diff
@@ -0,0 +1,121 @@
+[CmdletBinding(SupportsShouldProcess)]
+Param(
+    [Parameter(Mandatory=$true)][string]$AzCopyPath,
+    [Parameter(Mandatory=$true)][string]$BuildNumber,
+    [Parameter(Mandatory=$true)][string]$ReleaseVersion,
+    [Parameter(Mandatory=$true)][string]$DotnetStageAccountKey,
+    [Parameter(Mandatory=$true)][string]$DestinationAccountName,
+    [Parameter(Mandatory=$true)][string]$DestinationAccountKey,
+    [Parameter(Mandatory=$true)][string]$ChecksumsAccountName,
+    [Parameter(Mandatory=$true)][string]$ChecksumsAccountKey
+)
+
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 2.0
+
+$sourceAccountName = 'dotnetstage'
+$sourceContainerName = 'dotnet-monitor'
+$destinationContainerName = 'dotnet'
+
+function Generate-Source-Uri{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)][string]$AssetType
+    )
+
+    return "https://$sourceAccountName.blob.core.windows.net/$sourceContainerName/$BuildNumber/${AssetType}Assets/*"
+}
+
+function Generate-Destination-Uri{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)][string]$AccountName
+    )
+
+    return "https://$AccountName.blob.core.windows.net/$destinationContainerName/diagnostics/monitor/$ReleaseVersion"
+}
+
+function Generate-Sas-Token{
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)][string]$StorageAccountName,
+        [Parameter(Mandatory=$true)][string]$ContainerName,
+        [Parameter(Mandatory=$true)][string]$AccountKey,
+        [Parameter(Mandatory=$true)][string]$Permissions
+    )
+
+    $context = New-AzStorageContext -StorageAccountName $StorageAccountName -StorageAccountKey $AccountKey;
+    return New-AzStorageContainerSASToken -Container $ContainerName -Context $context -Permission $Permissions -ExpiryTime (Get-Date).AddHours(1.0);
+}
+
+function Transfer-File{
+    [CmdletBinding(SupportsShouldProcess)]
+    Param(
+        [Parameter(Mandatory=$true)][string]$From,
+        [Parameter(Mandatory=$true)][string]$To,
+        [Parameter(Mandatory=$true)][string]$FromToken,
+        [Parameter(Mandatory=$true)][string]$ToToken
+    )
+
+    Write-Verbose "Copy $From -> $To"
+
+    if ($From -eq $to) {
+        Write-Verbose 'Skipping copy because source and destination are the same.'
+    } else {
+        [array]$azCopyArgs = "$from$fromToken"
+        $azCopyArgs += "$to$toToken"
+        $azCopyArgs += "--s2s-preserve-properties"
+        $azCopyArgs += "--s2s-preserve-access-tier=false"
+        if ($WhatIfPreference) {
```

Comment:
What's the context of `WhatIf` and `WhatIfPreference`? Mostly just confused about the naming convention, so figured I'd ask

Reply:
WhatIf/$WhatIfPreference is the dry run concept but built into PowerShell.

If you specify the `-WhatIf` switch on a cmdlet that supports it (by adding `SupportsShouldProcess` to the `CmdletBinding` attribute), then the cmdlet should do all of the work it normally does EXCEPT for making a material impact. The cmdlet can inspect if `-WhatIf` was specified by checking the `$WhatIfPreference` variable which will be `$true` if it was specified and otherwise `$false`. 

You can read up on all of the details here: https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-shouldprocess?view=powershell-7.2